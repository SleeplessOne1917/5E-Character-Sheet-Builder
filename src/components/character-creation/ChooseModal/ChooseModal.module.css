.modal-container {
    background-color: rgba(03, 03, 03, 0.6);
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    z-index: 100;
    justify-content: center;
    align-items: center;
}

.modal {
    width: 100%;
    height: 100%;
    background-color: var(--heaven);
    display: grid;
    grid-template-columns: 5fr 2fr;
    grid-template-rows: 1fr 4fr 1fr;
    grid-template-areas: "icon icon"
                        "content content"
                        "buttons buttons";
    font-family: var(--font-display);
}

.content-container {
    grid-area: content;
    display: flex;
    flex-direction: column;
    align-items: center;
    overflow: auto;
    padding-bottom: 2rem;
    padding-top: 1rem;
}

.icon-container {
    padding: 1rem;
    grid-area: icon;
    display: flex;
    justify-content: center;
    align-items: center;
}

.icon {
    height: 10rem;
    width: 10rem;
}

.buttons-container {
    grid-area: buttons;
    display: flex;
}

.error-message {
    font-size: 1.3rem;
}

.modal-content-container {
    width: 100%;
    overflow-y: auto;
    display: flex;
    flex-direction: column;
    align-items: center;
    border-top: 3px solid var(--davy);
    height: 100%;
}

.title {
    color: var(--davy);
    font-family: var(--font-fantasy);
    font-size: 2rem;
}

@media screen and (min-width: 768px) {
    .modal {
        height: 80%;
        width: 80%;
    }
}

@media screen and (min-width:1024px) {
    .modal {
        height: 60%;
        width: 60%;
        grid-template-areas: "content icon"
                             "content ."
                             "buttons buttons";
        grid-template-rows: 3fr 3fr 1fr;
    }

    .icon {
        height: 20rem;
        width: 20rem;
    }
}