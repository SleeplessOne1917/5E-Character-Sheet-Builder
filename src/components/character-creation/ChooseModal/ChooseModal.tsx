import { KeyboardEventHandler, ReactNode, useCallback } from 'react';

import classes from './ChooseModal.module.css';

type ChooseModalProps = {
	show: boolean;
	mainContent: ReactNode;
	onClose: () => void;
	onChoose: () => void;
};

const ChooseModal = ({
	show,
	mainContent,
	onClose,
	onChoose
}: ChooseModalProps): JSX.Element => {
	const onCloseKeyUp: KeyboardEventHandler<HTMLDivElement> = useCallback(
		event => {
			if (event.code === 'Enter') {
				onClose();
			}
		},
		[onClose]
	);

	const onChooseKeyUp: KeyboardEventHandler<HTMLDivElement> = useCallback(
		event => {
			if (event.code === 'Enter') {
				onChoose();
			}
		},
		[onChoose]
	);

	return (
		<div
			style={{ display: show ? 'flex' : 'none' }}
			className={classes['modal-container']}
		>
			<div className={classes.modal}>
				<div className={classes['icon-container']}></div>
				<div className={classes['content-container']}>{mainContent}</div>
				<div className={classes['buttons-container']}>
					<div
						className={`${classes.button} ${classes.choose}`}
						tabIndex={0}
						aria-label="Choose"
						onClick={onChoose}
						onKeyUp={onChooseKeyUp}
					>
						Choose
					</div>
					<div
						className={`${classes.button} ${classes.cancel}`}
						tabIndex={0}
						aria-label="Cancel"
						onClick={onClose}
						onKeyUp={onCloseKeyUp}
					>
						Cancel
					</div>
				</div>
			</div>
		</div>
	);
};

export default ChooseModal;
