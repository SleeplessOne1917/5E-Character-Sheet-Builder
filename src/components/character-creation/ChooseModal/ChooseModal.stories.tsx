import ChooseModal, { ChooseModalProps } from './ChooseModal';
import { ComponentMeta, ComponentStory } from '@storybook/react';

export default {
	title: 'Components/ChooseModal',
	component: ChooseModal,
	args: { mainContent: 'Content', show: true },
	argTypes: {
		onChoose: { type: 'function' },
		onClose: { type: 'function' }
	}
} as ComponentMeta<typeof ChooseModal>;

export const Default: ComponentStory<typeof ChooseModal> = (
	args: ChooseModalProps
) => <ChooseModal {...args} />;
