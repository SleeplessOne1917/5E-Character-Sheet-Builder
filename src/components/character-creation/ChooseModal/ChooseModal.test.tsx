import '@testing-library/jest-dom';

import * as stories from './ChooseModal.stories';

import { render, screen } from '@testing-library/react';

import ChooseModal from './ChooseModal';
import { composeStories } from '@storybook/testing-react';
import userEvent from '@testing-library/user-event';

const { Default } = composeStories(stories);

it('renders correctly', () => {
	render(<Default />);

	expect(screen.getByTestId(/choose-modal/i)).toMatchSnapshot();
});

it('has display none when show is false', () => {
	render(<Default show={false} />);

	expect(screen.getByTestId(/choose-modal/i)).toHaveStyle('display: none');
});

it('has display flex when show is true', () => {
	render(<Default />);

	expect(screen.getByTestId(/choose-modal/i)).toHaveStyle('display: flex');
});

it('calls onChoose when choose button is clicked', async () => {
	const onChooseMock = jest.fn();

	render(
		<ChooseModal
			mainContent="foo"
			show
			onChoose={onChooseMock}
			onClose={() => {}}
		/>
	);

	await userEvent.click(screen.getByText(/choose/i));

	expect(onChooseMock).toHaveBeenCalled();
});

it('calls onClose when cancel button is clicked', async () => {
	const onCloseMock = jest.fn();

	render(
		<ChooseModal
			mainContent="foo"
			show
			onChoose={() => {}}
			onClose={onCloseMock}
		/>
	);

	await userEvent.click(screen.getByText(/cancel/i));

	expect(onCloseMock).toHaveBeenCalled();
});
