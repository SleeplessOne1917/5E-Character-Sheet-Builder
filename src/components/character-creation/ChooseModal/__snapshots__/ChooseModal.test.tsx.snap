// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="modal-container"
  data-testid="choose-modal"
  style="display: flex;"
>
  <div
    class="modal"
  >
    <div
      class="icon-container"
    />
    <div
      class="content-container"
    >
      Content
    </div>
    <div
      class="buttons-container"
    >
      <button
        class="button choose"
      >
        Choose
      </button>
      <button
        class="button cancel"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
